nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -72,7 +72,7 @@
 }
 
 
-class OneShotPruner(Pruner):
+class ToolBasedPruner(Pruner):
```

Comment:
what does "tool" refer to for this class? "tool" is a too general word

Reply:
also, add docstring for this class

Reply:
"tool" refer to `DataCollector`, `MetricsCalculator` and `SparsityAllocator`, or rename to `BasicPruner`?

Reply:
change to `BasicPruner`